processor/metricstransformprocessor/metrics_transform_processor.go

```diff
@@ -27,10 +29,42 @@ import (
 )
 
 type internalTransform struct {
-	MetricName string
-	Action     ConfigAction
-	NewName    string
-	Operations []internalOperation
+	MetricIncludeFilter internalFilter
+	Action              ConfigAction
+	NewName             string
+	Operations          []internalOperation
+}
+
+type internalFilter interface {
+	getMatches(toMatch metricNameMapping) []*metricspb.Metric
+}
+
+type internalFilterStrict struct {
+	include string
+}
+
+func (f internalFilterStrict) getMatches(toMatch metricNameMapping) []*metricspb.Metric {
+	if metrics, ok := toMatch[f.include]; ok {
+		return metrics
+	}
+
+	return nil
+}
+
+type internalFilterRegexp struct {
+	include *regexp.Regexp
+}
+
+func (f internalFilterRegexp) getMatches(toMatch metricNameMapping) []*metricspb.Metric {
+	const capExpectedMatches = 10
```

Comment:
Any reasoning for this value?

Reply:
Just seemed like a sensible default. If a user is using `regex` match type, they probably expect more than 1 match.

Reply:
Sounds good. Perhaps add this as a comment to the code? Otherwise, it's still going to be a magic number :)